AsyncNetworkService/Common/RequestLogger.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  RequestLogger.swift
+//  
+//
+//  Created by Paul Alvarez on 17/10/23.
+//
+
+import Foundation
+
+public final class RequestLogger {
+    public static let shared = RequestLogger()
+    private let queue = DispatchQueue(label: "com.dustidentity.RequestLoggerQueue")
```

Comment:
isn't this a background thread? i dont remember

isn't `DispatchQueue.main.async { ... }` for main thread? and anything else would be bg?

Reply:
Yes, that's a background thread, but the main issue was updating the value from different threads at the same time (it was done in an async method).
I think doing this operation on main thread would solve this too, but only because we are ensuring that it's executing in the same thread.
If you want I can make it run on main thread so we are confident that this will work

Reply:
ill test this now, ill let you know